web-scrapers/sqa_scraper.py

```diff
@@ -0,0 +1,276 @@
+    """
+    Fetches the create date and the date of correction of the dataset from the BeautifulSoup object.
+
+    Returns:
+        date (str): A string of dataset's create date.
+    """
+
+    fetched_create_date = "NULL"
+    fetched_update_date = "NULL"
+
+    part = page.find("a", string=ul.get_text())
+    # print("part", part, type(part), part.parent.parent.parent)
+    find_ul = part.find_parent("ul")
+    # print("1", part, part.find_parent("ul"), part.parent.parent.parent)
+    # print("2", part, part.find_parents("ul"), part.parent)
+    find_siblings = find_ul.next_siblings
+    for sibling in find_siblings:
+        if "<h2>" in repr(sibling):
+            # print("h2", repr(sibling))
+            return fetched_create_date, fetched_update_date
+        elif "Date of publication" in repr(sibling):
+            # print("date", repr(sibling))
+            fetched_create_date = sibling.get_text().split(":")[1].strip(" .\xa0&nbsp")
+            # print(fetched_create_date)
+        elif "Date of correction" in repr(sibling):
+            fetched_update_date = sibling.get_text().split(":")[1].strip(" .\xa0&nbsp")
+
+    return fetched_create_date, fetched_update_date
+
+
+def fetch_file_size(page: BeautifulSoup, ul: BeautifulSoup) -> tuple:
+    size = "NULL"
+    unit = "NULL"
+    part = page.find("a", string=ul.get_text())
+    # print("part", part, type(part), part.parent.parent.parent)
+    size_list = part.find_parent("li").contents
+    for item in size_list:
+        if "(" in item:
+            # print(item)
+            size = item.strip(" \xa0&nbsp()\n").split(" ")[0]
+            unit = item.strip(" \xa0&nbsp()\n").split(" ")[1]
+
+    return size, unit
+
+
+def fetch_description(ds, ys):
+    descr = "A range of statistical reports for SQA qualifications for " + ys.split(" ", 1)[1] + "."
+
+    """content = ds.find(id="content")
+    headlines = content.find_all(re.compile("h"))
+    for headline in headlines:
+        print(headline)
+        descr = headline.next_sibling()
+        descr_text= descr.get_text()
+    """
+    return descr
+
+
+if __name__ == "__main__":
+    # Record Headings
+    header = [
+        "Title",
+        "Owner",
+        "PageURL",
+        "AssetURL",
+        "DateCreated",
+        "DateUpdated",
+        "FileSize",
+        "FileSizeUnit",
+        "FileType",
+        "NumRecords",
+        "OriginalTags",
+        "ManualTags",
+        "License",
+        "Description",
+    ]
+    data = []
+
+    print("Getting available years")
+    category_links = fetch_available_years()
+    for year_string in category_links.keys():
+        print("Getting", year_string)
+        years_page = fetch_year_page(category_links[year_string])
+        owner = "Scottish Qualifications Authority (SQA)"
+        title = create_title(year_string)
+        # print("title", title)
+        description = fetch_description(years_page, year_string)
+        print(description)
```

Comment:
scraper is very verbose. I would recommend removing the print statement for `description` as it's not any more information than the print statement in line 225.